test/jdk/javax/swing/JTableHeader/JTableHeaderLabelRightAlignTest.java

```diff
@@ -0,0 +1,101 @@
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import javax.imageio.ImageIO;
+import javax.swing.JLabel;
+import javax.swing.JTable;
+import javax.swing.SwingUtilities;
+import javax.swing.table.JTableHeader;
+import javax.swing.UIManager;
+
+import java.awt.Dimension;
+import java.awt.Graphics2D;
+import java.awt.image.BufferedImage;
+
+import java.io.File;
+import java.io.IOException;
+
+/*
+ * @test
+ * @bug 8301606
+ * @summary Test to check if the Right aligned header
+ *          label doesn't cut off Metal Look&Feel
+ * @run main JTableHeaderLabelRightAlignTest
+ */
+public class JTableHeaderLabelRightAlignTest {
+    private static final int WIDTH = 300;
+    private static final int HEIGHT = 150;
+    private static final double SCALE = 2.25;
+
+    public static void main(String[] args) throws Exception {
+        UIManager.setLookAndFeel("javax.swing.plaf.metal.MetalLookAndFeel");
+        SwingUtilities.invokeAndWait(JTableHeaderLabelRightAlignTest::test);
+    }
+
+    private static void test() {
+        String[][] data = {
+                {"1", "1"}
+        };
+
+        String[] columnNames = {"Size", "Size"};
+
+        JTable table = new JTable(data, columnNames);
+        table.setSize(WIDTH, HEIGHT);
+        ((JLabel)table.getTableHeader().getDefaultRenderer())
+                .setHorizontalAlignment(JLabel.RIGHT);
+
+        final JTableHeader header = table.getTableHeader();
+        Dimension size = header.getPreferredSize();
+        header.setSize(size);
+
+        int w = (int)Math.ceil(SCALE * size.width);
+        int h = (int)Math.ceil(SCALE * size.height);
+        BufferedImage imgHeader = new BufferedImage(w, h,
+                BufferedImage.TYPE_INT_RGB);
+        Graphics2D g2d = imgHeader.createGraphics();
+        g2d.scale(SCALE, SCALE);
+        try {
+            header.paint(g2d);
+        } finally {
+            g2d.dispose();
+        }
+
+        int x = (int)(table.getTableHeader().
+                getColumnModel().getColumn(0).getWidth() * SCALE);
+        int expectedRGB = imgHeader.getRGB(x, 1);
+
+        for (int y = 1; y < (imgHeader.getHeight() - 3); y++) {
+            if (expectedRGB != imgHeader.getRGB(x, y)) {
+                try {
+                    ImageIO.write(imgHeader, "png",
+                            new File("FailureImage.png"));
+                } catch (IOException ioException) {
+                    ioException.printStackTrace();
+                }
+                throw new RuntimeException("Test Failed at <" +
+                        x + ", " + y + ">");
```

Comment:
```suggestion
                throw new RuntimeException("Test Failed at <" + x + ", " + y + ">");
```
It doesn't fit 80-column limit, but I think it is okay to keep it on one line without wrapping. It doesn't affect the readability of the code, and it's only 5 chars above the limit.